src/ObservationDims.jl

```diff
@@ -1,5 +1,159 @@
 module ObservationDims
 
-greet() = print("Hello World!")
+using NamedDims
+using AxisArrays
+using Distributions
+
+export obs_arrangement, organise_obs
+export SingleObs, IteratorOfObs, ArraySlicesofObs
+export MatrixRowsOfObs, MatrixColsOfObs
+
+"""
+    type ObsArrangement
+
+Defines the orientation of data that is expected by a function
+"""
+abstract type ObsArrangement end
+
+struct SingleObs <: ObsArrangement end
+struct IteratorOfObs <: ObsArrangement end
+struct ArraySlicesOfObs{D} <: ObsArrangement end
+
+const MatrixRowsOfObs = ArraySlicesOfObs{1}
+const MatrixColsOfObs = ArraySlicesOfObs{2}
+
+
+"""
+    organise_obs(f, data; obsdim=nothing)
+
+Organise the `data` according to the `ObsArrangement` expected by some function `f`.
+"""
+function organise_obs(f, data; obsdim=nothing)
+    return organise_obs(obs_arrangement(f), data; obsdim=obsdim)
+end
+
+
+# Specify arrangement based on type of data:
+
+# Scalars have no "orientation" so no rearrangement required
+for T in (Sampleable, Number, Symbol)
+    @eval organise_obs(f, data::$T; obsdim=nothing) = data
+    @eval organise_obs(::SingleObs, data::$T; obsdim=nothing) = data
+end
```

Comment:
why do we need these "Scalar" definitions?

Is it not sufficient to have (as below) `organise_obs(::SingleObs, data; obsdim=nothing) = data`

Reply:
remember we got an ambiguity error because `organise_obs(::SingleObs, data::$T; obsdim=nothing)` wasn't defined?

Reply:
Right, but why do we need either? We'd not get a ambiguity if the first of these definitions weren't here too

I guess let's open an issue to discuss if organise_obs on a scalar-like (e.g. number) should requires SingleObs

Reply:
>We'd not get a ambiguity if the first of these definitions weren't here too

since we only have 4 ObsArrangements I made these declarations more explicit

Reply:
yeah, my question (which i'll open as an issue) is: do the `IteratorOfObs` and `ArraySlicesOfObs` methods make sense?

Reply:
https://github.com/invenia/ObservationDims.jl/issues/11